### ISSUE TITLE
Wide discrepancy on read times on files of similar size

### ISSUE THREAD

**Issue by xiaodaigh** [bicycle1885/TableReader.jl#36]
This read in 12 seconds on my computer 
```julia
download("https://github.com/xiaodaigh/testing/raw/master/Performance_2016Q1.zip", "ok.zip")
run(`unzip -o ok.zip`)
using TableReader
path = "Performance_2016Q1.csv"
@time a = readcsv(path, delim = '|', hasheader = false); # 12~15 seconds
```

but this takes 45 seconds and the file is only 200mb larger than the above one. Why is this? Does it have to do with missing values? (files being uploaded to github)

```julia
download("https://github.com/xiaodaigh/testing/raw/master/Performance_2000Q1.zip", "ok2.zip")
run(`unzip -o ok2.zip`)
using TableReader
path = "Performance_2000Q1.txt"
@time a2 = readcsv(path, delim = '|', hasheader = false); # 45seconds
```

**Comment by bicycle1885** [comment]
Thank you for reporting it. I'll take a look soon.

**Comment by bicycle1885** [comment]
I think this is fixed by #37, which made reading of the pathological case ~3-4 times faster.
